utils/cmake/dreamcast.toolchain.cmake

```diff
@@ -0,0 +1,106 @@
+# CMake Toolchain file for targeting the Dreamcast or NAOMI with CMake.
+#  (c)2023 Falco Girgis, Luke Benstead
+#
+# This file is to be passed to CMake when compiling a regular CMake project
+# to cross-compile for the Dreamcast, using the KOS environment and settings.
+#
+#     cmake /path/to/src -DCMAKE_TOOLCHAIN_FILE=${KOS_BASE}/kos/utils/cmake/dreamcast.toolchain.cmake
+#
+#   or alternatively:
+#
+#     cmake /path/to/src -DCMAKE_TOOLCHAIN_FILE=${KOS_CMAKE_TOOLCHAIN}
+#
+# The toolchain enables the options required for the compiler to use the 
+# special SH4 instructions for sin/cos and invSqrt. Note that even when 
+# these options are present, they are not activated unless -ffast-math 
+# is also provied. Typically you will want to provide a project-level 
+# option for enabling this flag, which will make the toolchain's SH4 
+# instruction flags also take effect.
+#
+# Frame pointers are enabled in debug builds as these are required for 
+# stack traces and GDB. They are disabled in release.
+# 
+# The original toolchain file was created by Kazade for the Simulant 
+# engine who has graciously allowed the rest of the scene to warez it. 
+#
+# Special thanks to Matt Godbolt of Compiler Explorer, for supporting
+# the SH toolchains, which allowed us to figure out the precise magical
+# incantations of compiler flags for fast math.  
+
```

Comment:
We should add checks here to make sure the ENV vars we use are defined, otherwise you get errors from incorrect paths when we could display meaningful errors.

```
#### Check Required Environment Variables ####
if(NOT DEFINED ENV{KOS_BASE})
  message(FATAL_ERROR "Environment Variable KOS_BASE not set")
endif()

if(NOT DEFINED ENV{KOS_CC_BASE})
  message(FATAL_ERROR "Environment Variable KOS_CC_BASE not set")
endif()

if(NOT DEFINED ENV{KOS_SUBARCH})
  message(FATAL_ERROR "Environment Variable KOS_CC_BASE not set")
endif()

if(NOT DEFINED ENV{KOS_PORTS})
  message(FATAL_ERROR "Environment Variable KOS_CC_BASE not set")
endif()
```

Reply:
Also agreed, that's cleaner and would let you manually set them.